hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -92,28 +104,26 @@ public void markBlocksAsDeleted(KeyValueContainerData data, SortedSet<Long> dele
           .clearDeletedBlocks()
           .addAllDeletedBlocks(sortedDeletedBlockIDs)
           .build();
-      write(data, checksumInfoBuilder.build());
+      captureLatencyNs(metrics.getUpdateContainerMerkleTreeLatencyNS(),
+          () -> write(data, checksumInfoBuilder.build()));
       LOG.debug("Deleted block list for container {} updated", data.getContainerID());
+    } catch (IOException ex) {
+      metrics.incrementMerkleTreeUpdateFailures();
+      throw new IOException("Error occurred when updating container merkle tree for containerID "
+          + data.getContainerID(), ex);
     } finally {
       writeLock.unlock();
     }
   }
 
-  public ContainerDiff diff(KeyValueContainerData thisContainer, File otherContainerTree)
+  public ContainerDiff diff(KeyValueContainerData thisContainer, ContainerProtos.ContainerChecksumInfo otherInfo)
```

Comment:
Let's leave this interface unchanged for now. The current plan is that diff will be passed whatever file or bytestream is obtained from the wire and deserialization into a proto will happen within this class.